### PR TITLE
DOC: Make input/output type consistent and add more examples for documentation

### PR DESCRIPTION
**Pull request by Sarah-Yifei-Wang**
I noticed in quite a few scimath function, the input type is array_like, but output type is ndarray or scalar. Initially this caused some confusion for me, making me feel input does not contain scale. To avoid future confusion, this PR is to:
1. make input type and output type consistent using array_like
2. add examples using scalar as input

### CODE REVIEW

**Comment by Sarah-Yifei-Wang** [comment]
Wonder if anyone knows how to fix the failed tests related to old_gcc and armv7_simd_test? Thanks in advance!

**Review by rossbar — COMMENTED** [review]
There's no need for the updated parameter descriptions, [`array_like` encompasses scalars](https://numpy.org/devdocs/glossary.html#term-array_like).

**Comment by Sarah-Yifei-Wang** [comment]
Got it! Thanks for the explanation. It looks like sometimes the input type uses "array_like" but output type uses "ndarray or scalar", I think that's what caused the confusion. I can update to make them consistent!

File: numpy/lib/scimath.py [hunk 1]
```diff
@@ -403,7 +403,7 @@ def log2(x):
 
     Returns
     -------
-    out : ndarray or scalar
+    out : array_like
```
Comment (rossbar) [inline numpy/lib/scimath.py:406]: In my opinion, this change would actually make the docs less clear as this makes the return type less specific. `array_like` is typically used for *inputs* to indicate that objects which *can* be treated as arrays (or scalars) by numpy *will be* internal to the function. Numpy functions almost always return arrays or scalars but not other sequence types (like lists) so `ndarray or scalar` is actually a better description for the return type IMO.

File: numpy/lib/scimath.py [hunk 2]
```diff
@@ -472,11 +472,18 @@ def power(x, p):
...
     array([0.25  ,  0.0625])
+
     >>> np.emath.power([-2, 4], 2)
     array([ 4.-0.j, 16.+0.j])
 
+    >>> np.emath.power([2,4], [2,4])
+    array([ 4, 256])
+
+    >>> np.emath.power(2, 2)
+    4
```
Comment (mdhaber) [inline numpy/lib/scimath.py:486]:
```suggestion
    >>> np.emath.power([2, 4], [2, 4])
    array([ 4, 256])

```

File: numpy/lib/scimath.py [hunk 3]
```diff
@@ -472,11 +472,18 @@ def power(x, p):
 
     >>> np.emath.power([2, 4], 2)
```
Comment (mdhaber) [inline numpy/lib/scimath.py:473]: Let's start with the simplest.
```suggestion
    >>> np.emath.power(2, 2)
    4
    
    >>> np.emath.power([2, 4], 2)
```

File: numpy/lib/scimath.py [hunk 4]
```diff
@@ -592,7 +599,7 @@ def arctanh(x):
 
     Returns
     -------
-    out : ndarray or scalar
+    out : array_like
```
Comment (mdhaber) [inline numpy/lib/scimath.py:602]:
```suggestion
    out : ndarray or scalar
```

**Review by mdhaber — APPROVED** [review]
The examples in which the output is a scalar LGTM. I'll go ahead and merge if CI looks good. Thanks @Sarah-Yifei-Wang.

**Pull request merged**